src/transformers/generation/utils.py

```diff
@@ -1635,7 +1635,8 @@ def get_layer_device_map(execution_device_map: Optional[dict] = None):
             # This is needed here if we don't want to make changes in accelerate in order to save execution_device
             # For offloaded case, we need to get the execution device, not just the device where it is offloaded
             if hasattr(self, "hf_device_map"):
-                main_device = [d for d in self.hf_device_map.values() if d not in ["cpu", "disk"]][0]
+                main_device = [d for d in self.hf_device_map.values() if d not in ["cpu", "disk"]]
+                main_device = main_device[0] if main_device else "cpu"
```

Comment:
What about this ? This is taken from the `dispatch_model` function. 
```suggestion
                if set(self.hf_device_map.values()) == {"cpu"} or set(self.hf_device_map.values()) == {"cpu", "disk"}:
                    main_device = "cpu"
                else:
                    main_device = [d for d in self.hf_device_map.values() if d not in ["cpu", "disk"]][0]
```